internal/repository/db.go

```diff
@@ -1,38 +1,50 @@
 package repository
 
 import (
-	"database/sql"
+	"context"
 	"strings"
 
 	"github.com/andreipimenov/golang-training-2021/internal/model"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
 type DB struct {
-	*sql.DB
+	*mongo.Collection
 }
 
-func NewDB(db *sql.DB) *DB {
+func NewDB(db *mongo.Collection) *DB {
 	return &DB{db}
 }
 
 func (db *DB) Load(key string) (model.Price, bool) {
-	var open, high, low, close string
 	ticker, date := splitKey(key)
-	err := db.QueryRow("SELECT open, high, low, close FROM prices WHERE price_date = $1 AND ticker = $2", date, ticker).Scan(&open, &high, &low, &close)
+	filter := bson.D{{Key: "ticker", Value: ticker}, {Key: "price_date", Value: date}}
+
+	var result model.Price
+	err := db.FindOne(context.TODO(), filter).Decode(&result)
 	if err != nil {
 		return model.Price{}, false
 	}
 	return model.Price{
```

Comment:
I do not fully understand why it was necessary to return a new variable. There is already a variable result of the required type and with the required data inside.

Reply:
Yes. Thanks.
There is better to use existing result variable in return statement.